### PR TITLE
Update dependency to github.com/cenkalti/backoff/v4

### PR DESCRIPTION
**Pull request by mikelolasagasti**
Update from `github.com/cenkalti/backoff/v3` to `github.com/cenkalti/backoff/v4`.

### CODE REVIEW

**Review by cipherboy — APPROVED** [review]
Thanks @mikelolasagasti!

**Comment by hixichen** [comment]
can not find log from:
https://github.com/openbao/openbao/actions/runs/9581745630/job/26419226564?pr=287

**Comment by cipherboy** [comment]
@hixichen Likely because this job ran so long ago. We'll need to rebase or recreate if we want to merge.

File: go.mod [hunk 1]
```diff
@@ -210,7 +210,7 @@ require (
 	github.com/beorn7/perks v1.0.1 // indirect
 	github.com/bgentry/speakeasy v0.1.0 // indirect
 	github.com/boombuler/barcode v1.0.1 // indirect
-	github.com/cenkalti/backoff/v4 v4.2.1 // indirect
+	github.com/cenkalti/backoff/v3 v3.2.2 // indirect
```
Comment (cipherboy) [inline go.mod:213]: Wonder what brings this in as indirect?

Comment (cipherboy) [reply go.mod:213]: Ah:

```
# github.com/cenkalti/backoff/v3
github.com/openbao/openbao/command
github.com/openbao/openbao-template/config
github.com/openbao/openbao/api
github.com/cenkalti/backoff/v3
```

So we'll have to update `openbao-template` to use a new API tag after this is released in order to drop the `/v3` fully. That's fine!

Comment (cipherboy) [reply go.mod:213]: @JanMa Something to remember to do for v2.1.0!

**Comment by cipherboy** [comment]
Thank you @mikelolasagasti! Sorry about the delay. :-)

**Pull request merged**